frontend/packages/dev-console/src/components/add/AddCardItem.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { useUserSettings } from '@console/shared';
+import { SimpleListItem, Title, Text } from '@patternfly/react-core';
+import { ResolvedExtension, AddAction } from '@console/dynamic-plugin-sdk';
+import { useTelemetry } from '@console/shared/src/hooks/useTelemetry';
+import { navigateTo, resolvedHref } from '../../utils/add-page-utils';
+import './AddCardItem.scss';
+
+type AddCardItemProps = {
+  action: ResolvedExtension<AddAction>;
+  namespace: string;
+};
+
+const AddCardItem: React.FC<AddCardItemProps> = ({
+  action: {
+    properties: { id, label, icon, href, description },
+  },
+  namespace,
+}) => {
+  const fireTelemetryEvent = useTelemetry();
+  const [showDetails] = useUserSettings('console.addPage.showDetails', 'show', true);
+  const actionIcon = (): JSX.Element => {
```

Comment:
How so? What are you thinking here @sahil143 ? This is just a more-complex ternary and we render those without memo all the time.

Reply:
Since this is a method that is being executed during render. we can use the hook use memo which will return the element/component that is needed to render
```
const actionIcon = React.useMemo(() => {
     if (typeof icon === 'string') {
      return (
        <img
          className="odc-add-card-item__header__icon"
          src={icon}
          alt={label}
          aria-hidden="true"
        />
      );
    }
    if (typeof icon !== 'string' && React.isValidElement(icon)) {
      return (
        <span className="odc-add-card-item__header__icon" aria-hidden="true">
          {icon}
        </span>
      );
    }
    return null;
}, [icon, label])
```

Reply:
IMHO, there is not much gain here at the cost of using useMemo, since there is no resource intensive computation.